File: csharp/atbash-cipher/AtbashCipher.cs
public static class AtbashCipher
{
    public static string Encode(string plainValue) => string.Join(' ', Translate(plainValue)
            .Chunk(5)
            .Select(chunk => new string(chunk)));

    public static string Decode(string encodedValue) => Translate(encodedValue);

    private static string Translate(string s) => string.Join("", s.ToLower()
        .Where(char.IsLetterOrDigit)
        .Select(x =>
        {
            return char.IsLetter(x)
                ? (char)('z' - (x - 'a'))
                : x;
        }));
}

File: cpp/atbash-cipher/atbash_cipher.cpp
#include "atbash_cipher.h"

namespace atbash_cipher
{
    static std::string translate(const std::string &s)
    {
        std::string result;

        for (unsigned char ch : s)
        {
            ch = std::tolower(ch);

            if (std::isalnum(ch))
            {
                if (std::isalpha(ch))
                {
                    result += static_cast<char>('z' - (ch - 'a'));
                }
                else
                {
                    result += ch;
                }
            }
        }

        return result;
    }

    std::string encode(const std::string &plainValue)
    {
        std::string translated = translate(plainValue);
        std::string result;

        for (size_t i = 0; i < translated.size(); ++i)
        {
            if (i > 0 && i % 5 == 0)
            {
                result += ' ';
            }

            result += translated[i];
        }

        return result;
    }

    std::string decode(const std::string &cipher)
    {
        return translate(cipher);
    }

} // namespace atbash_cipher
